bpm/bonita-util/bonita-common-util/src/test/java/org/bonitasoft/engine/bdm/BusinessObjectModelConverterTest.java

```diff
@@ -80,10 +79,25 @@ public void unzipADifferentZipThrowAnException() throws Exception {
 
     @Test
     public void should_be_backward_compatible() throws Exception {
-        byte[] xml = org.apache.commons.io.IOUtils.toByteArray(BusinessObjectModelConverterTest.class.getResourceAsStream("/bom_6.3.0.xml"));
-        BusinessObjectModelConverter convertor = new BusinessObjectModelConverter();
-        convertor.unmarshall(xml);
+        final byte[] xml = org.apache.commons.io.IOUtils.toByteArray(BusinessObjectModelConverterTest.class.getResourceAsStream("/bom_6.3.0.xml"));
+        final BusinessObjectModelConverter convertor = new BusinessObjectModelConverter();
+        final BusinessObjectModel bom = convertor.unmarshall(xml);
         // expect no unmarshalling exception
+
+        assertThat(bom.getModelVersion()).isNullOrEmpty();
+        assertThat(bom.getProductVersion()).isNullOrEmpty();
+    }
+
+    @Test
+    public void should_set_current_model_version_when_marshalling() throws Exception {
+        final BusinessObjectModelConverter convertor = new BusinessObjectModelConverter();
+        final BusinessObjectModel bom = new BusinessObjectModelBuilder().buildBOMWithQuery();
+        assertThat(bom.getModelVersion()).isNullOrEmpty();
+        assertThat(bom.getProductVersion()).isNullOrEmpty();
+        final BusinessObjectModel unmarshalledBom = convertor.unmarshall(convertor.marshall(bom));
+
+        assertThat(unmarshalledBom.getModelVersion()).isEqualTo(BusinessObjectModel.CURRENT_MODEL_VERSION);
+        assertThat(bom.getProductVersion()).isNotNull().isNotEmpty();
```

Comment:
it should compare product version against pom project version.
this could be a migration step check